ls1intum/Artemis
# `Discussion`: Reduce loading time of discussion page with many posts by using pagination

### Checklist
#### General


#### Server
#### Client
- [x] I translated all newly inserted strings into English and German (With translation service, please check). 

### Motivation and Context


Resolves: #4433
### Description
Pagination has been added to course discussion page. All filtering, sorting operations moved to backend server to enhance the performance.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Log in to Artemis
2. Navigate to Course Discussions Page
3. Verify whether filtering/sorting of posts on page occur as expected.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage




AnswerPostService.java 100% | 98%
PostService.java 100% | 78%
ReactionService 100% | 100% 


course-discussion.component.ts: 82.55 %|   95.65%
### Screenshots


![image](https://user-images.githubusercontent.com/67344158/153228025-168dd447-379d-4d8d-868b-3c26138dab91.png)

https://user-images.githubusercontent.com/67344158/153227491-5d35c827-4276-4b03-9ba7-4a62b3c21f94.mp4
